src/Utilities/RootSolvers/src/RootSolvers.jl

```diff
@@ -2,162 +2,129 @@
-x_star2 = 10000.0
-args = Tuple(x_star2)
-x_star = sqrt(x_star2)
-tol_abs = 1.0e-3
-iter_max = 100
-x_root, converged = find_zero.(f,
-                               x_0,
-                               x_1,
-                               Ref(args),
-                               Ref(IterParams(tol_abs, iter_max)),
-                               Ref(SecantMethod()))
+x_root, converged = find_zero(x -> x^2 - 100^2, 0.0, 1000.0, SecantMethod())
 ```
 
 """
 module RootSolvers
 
-export find_zero
-export IterParams
-
-export SecantMethod
-export RegulaFalsiMethod
+export find_zero, SecantMethod, RegulaFalsiMethod, NewtonsMethod
 
-struct IterParams{R<:AbstractFloat, I<:Int}
-  tol_abs::R
-  iter_max::I
-end
+import ForwardDiff
 
 abstract type RootSolvingMethod end
-struct SecantMethod<:RootSolvingMethod end
-struct RegulaFalsiMethod<:RootSolvingMethod end
+Base.broadcastable(method::RootSolvingMethod) = Ref(method)
 
-get_solve_methods() = (SecantMethod(), RegulaFalsiMethod())
+struct SecantMethod <: RootSolvingMethod end
+struct RegulaFalsiMethod <: RootSolvingMethod end
+struct NewtonsMethod <: RootSolvingMethod end
 
 """
-`find_zero(f, x_0, x_1, args, iter_params, SecantMethod)`
-
-Solves the root equation, `f`, using Secant method.
-See [RootSolvers](@ref) for more information.
+    x, converged = find_zero(f, x0[, x1], method;
+                             xatol=0, xrtol=sqrt(eps(eltype(x0))), 
+                             yatol=sqrt(eps(eltype(x0))), maxiters=10_000)
+
+Finds the nearest root of `f` to `x0` and `x1`. Returns a the value of the root `x` such
+that `f(x) ≈ 0`, and a Boolean value `converged` indicating convergence.
+
+`method` can be one of:
+- `SecantMethod()`: [Secant method](https://en.wikipedia.org/wiki/Secant_method)
+- `RegulaFalsiMethod()`: [Regula Falsi method](https://en.wikipedia.org/wiki/False_position_method#The_regula_falsi_(false_position)_method).
+- `NewtonsMethod()`: [Newton's method](https://en.wikipedia.org/wiki/Newton%27s_method)
+  - The `x1` argument is omitted for Newton's method.
+
+The keyword arguments:
+- `xatol` and `xrtol` are the absolute and relative tolerances of the input.
+- `yatol` is the absolute tolerance of the output of `f`.
+- `maxiters` is the maximum number of iterations.
 """
-function find_zero(f::F,
-                   x_0::T,
-                   x_1::T,
-                   args::Tuple,
-                   iter_params::IterParams{R, Int},
-                   method::SecantMethod
-                   )::Tuple{T, Bool} where {F, R, T <: Union{R, AbstractArray{R}}}
-  iter_max = iter_params.iter_max
-  tol_abs = iter_params.tol_abs
-  Δx = x_1 - x_0
-  y_0 = f(x_0, args...)
-  y_1 = f(x_1, args...)
-  x = x_1 - y_1 * Δx / (y_1 - y_0) # puts x in outer scope, & saves iter_max = 0 case
-  for i in 1:iter_max
-    x = x_1 - y_1 * Δx / (y_1 - y_0)
-    y_0 = y_1
-    x_0 = x_1
-    x_1 = x
-    y_1 = f(x_1, args...)
-    Δx = x_1 - x_0
-    if abs(y_1) < tol_abs
+function find_zero end
```

Comment:
What's this for/do?

Reply:
basically it defines a function without any methods: it's handy for adding docs where you want the same docs to be shared by different methods.